core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceMultipartTest.java

```diff
@@ -89,6 +90,25 @@ void testUploadFile(String path) throws Exception {
                            "\"param1\":\"armeria\"}");
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = "/uploadWithFileParam")
+    void testUploadFileWithUnexpectedParameters(String path) throws Exception {
+        final Multipart multipart = Multipart.of(
+                BodyPart.of(ContentDisposition.of("form-data", "file1", "foo.txt"), "foo"),
+                BodyPart.of(ContentDisposition.of("form-data", "path1", "bar.txt"), "bar"),
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile1", "qux.txt"), "qux"),
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile2", "quz.txt"),
+                            MediaType.PLAIN_TEXT, "quz"),
+                // Unexpected parameter: multipartFile3
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile3", "qux3.txt"), "qux3"),
+                BodyPart.of(ContentDisposition.of("form-data", "param1"), "armeria")
+
+        );
+        final AggregatedHttpResponse response =
+                server.blockingWebClient().execute(multipart.toHttpRequest(path));
+        assertEquals(HttpStatus.OK, response.status());
```

Comment:
This doesn't make sure the `multipartFile3` has been either discarded or processed at all. What about writing a new endpoint in `MyAnnotatedService` that makes sure the file is not created for `multipartFile3` in the directory that contains `multipartFile1` and `multipartFile2`.